browser/brave_tab_helpers.cc

```diff
@@ -135,9 +136,14 @@ void AttachTabHelpers(content::WebContents* web_contents) {
           return skus::SkusServiceFactory::GetForContext(context);
         },
         context);
+    if (!g_brave_browser_process->process_misc_metrics()) {
```

Comment:
I think probably it makes more sense to CHECK_IS_TEST  for nullptr in `AIChatTabHelper::CreateForWebContents`

Reply:
yes, looks like it already handles null metrics internally

Reply:
Removed in https://github.com/brave/brave-core/pull/20433/commits/1f9360f2f6b55f1e954a10aa9365ce174cbc8b66